presto-iceberg/src/main/java/com/facebook/presto/iceberg/statistics/KllHistogram.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    public static boolean isKllHistogramSupportedType(Type type)
+    {
+        try {
+            return isNumericType(type) ||
+                    type instanceof AbstractIntType;
+        }
+        catch (PrestoException e) {
+            return false;
+        }
+    }
+
+    @JsonProperty
+    public Slice getSketch()
+    {
+        return Slices.wrappedBuffer(sketch.toByteArray());
+    }
+
+    @JsonProperty
+    public Type getType()
+    {
+        return type;
+    }
+
+    @VisibleForTesting
+    @SuppressWarnings("rawtypes")
+    public KllItemsSketch getKllSketch()
+    {
+        return sketch;
+    }
+
+    @Override
+    public Estimate cumulativeProbability(double value, boolean inclusive)
+    {
+        return Estimate.of(sketch.getRank(fromDouble.apply(value), inclusive ? INCLUSIVE : EXCLUSIVE));
+    }
+
+    @Override
+    public Estimate inverseCumulativeProbability(double percentile)
+    {
+        return Estimate.of(toDouble.apply(sketch.getQuantile(percentile)));
+    }
+
+    /**
+     * The memory utilization is dominated by the size of the sketch. This estimate
+     * doesn't account for the other fields in the class.
+     */
+    @Override
+    public long getEstimatedSize()
+    {
+        return INSTANCE_SIZE + sketch.getSerializedSizeBytes();
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("type", type)
+                .add("k", this.sketch.getK())
+                .add("N", this.sketch.getN())
+                .add("retained", this.sketch.getNumRetained())
+                .add("mingetSerialized", this.sketch.getMinItem())
+                .add("max", this.sketch.getMaxItem())
+                .add("p50", sketch.getQuantile(0.5))
+                .add("p75", sketch.getQuantile(0.75))
+                .add("p90", sketch.getQuantile(0.90))
+                .add("p99", sketch.getQuantile(0.99))
+                .add("p99.9", sketch.getQuantile(0.999))
+                .toString();
+    }
+
+    private static class SketchParameters<T>
+    {
+        private final Comparator<T> comparator;
+        private final ArrayOfItemsSerDe<T> serde;
+        private final Function<Object, Object> conversion;
+
+        public SketchParameters(Comparator<T> comparator, ArrayOfItemsSerDe<T> serde, Function<Object, Object> conversion)
+        {
+            this.comparator = comparator;
+            this.serde = serde;
+            this.conversion = conversion;
+        }
+
+        public SketchParameters(Comparator<T> comparator, ArrayOfItemsSerDe<T> serde)
+        {
+            this(comparator, serde, Function.identity());
+        }
+
+        public Comparator<T> getComparator()
+        {
+            return comparator;
+        }
+
+        public ArrayOfItemsSerDe<T> getSerde()
+        {
+            return serde;
+        }
+
+        public Function<Object, Object> getConversion()
```

Comment:
This is unused.. Was this the converter you actually wished to use before switching to toDouble/fromDouble. Should we still be setting `conversion` ?

Reply:
this was an artifact from a previous iteration -- I've removed it